src/language-js/comments.js

```diff
@@ -942,6 +947,46 @@ function handleTSMappedTypeComments(
   return false;
 }
 
+function hasQuestionBetweenTestAndComment(testNode, comment, text, options) {
+  const testNodeLocEnd = options.locEnd(testNode);
+  const commentLocStart = options.locStart(comment);
+  for (let i = testNodeLocEnd; i < commentLocStart; i++) {
```

Comment:
Why is the loop needed? Isn't one call to `getNextNonSpaceNonCommentCharacterIndexWithStartIndex` enough here? What else can there be between the test node and the comment?